src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -444,7 +444,26 @@ private void ScanSyntaxToken(ref TokenInfo info)
                     if (!this.ScanNumericLiteral(ref info))
                     {
                         TextWindow.AdvanceChar();
-                        info.Kind = SyntaxKind.DotToken;
+                        if (TextWindow.PeekChar() == '.')
+                        {
+                            TextWindow.AdvanceChar();
+                            if (TextWindow.PeekChar() == '.')
+                            {
+                                // Triple-dot: explicitly reject this, to allow triple-dot
+                                // to be added to the language without a breaking change.
+                                // (without this, 0...2 would parse as (0)..(.2), i.e. a range from 0 to 0.2)
+                                TextWindow.Reset(startingPosition);
```

Comment:
why would this be necessary?  is it to avoid ```a...b``` being parsed as ```(a..).b```?   Is that because "a.." will be allowed as some sort of open range?

Reply:
d'oh.  just saw the comment.